File: src/main/java/com/machfour/macros/core/AttrMapping.java
package com.machfour.macros.core;

import com.machfour.macros.data.ColumnData;
import com.machfour.macros.data.Table;
import com.machfour.macros.data.Schema;

public class AttrMapping extends MacrosEntity<AttrMapping> {

    public AttrMapping(ColumnData<AttrMapping> data, ObjectSource objectSource) {
        super(data, objectSource);
    }

    @Override
    public Table<AttrMapping> getTable() {
        return Schema.AttrMappingTable.instance();
    }

}

File: libmacros/src/main/kotlin/com/machfour/macros/entities/AttrMapping.kt
package com.machfour.macros.entities

import com.machfour.macros.sql.entities.Factory
import com.machfour.macros.sql.entities.MacrosEntityImpl
import com.machfour.macros.core.ObjectSource
import com.machfour.macros.sql.entities.Factories
import com.machfour.macros.schema.AttrMappingTable
import com.machfour.macros.sql.Table
import com.machfour.macros.sql.rowdata.RowData

class AttrMapping internal constructor(data: RowData<AttrMapping>, objectSource: ObjectSource)
    : MacrosEntityImpl<AttrMapping>(data, objectSource) {

    companion object {
        val factory: Factory<AttrMapping, AttrMapping>
            get() = Factories.attributeMapping

    }

    override fun getTable(): Table<*, AttrMapping> {
        return AttrMappingTable
    }

}
